lib/JsonNode.cpp

```diff
@@ -214,6 +214,49 @@ bool JsonNode::isNumber() const
 	return type == DATA_INTEGER || type == DATA_FLOAT;
 }
 
+bool JsonNode::isEmpty() const
+{
+	switch(type)
+	{
+		case DATA_NULL:
+			return true;
+		case DATA_STRUCT:

```

Comment:
This should handle vector type node too.

Reply:
I don't consider a vector with no elements to be "empty" on purpose here. The reason is that the intersection of two json nodes is the empty vector (rather than null) only if both are the empty vector. So when constructing a "base" object as intersection of nodes we want to preserve [ ] in the intersection, but not { } which (typically) will be the result of intersecting two distinct structs.
The name "isEmpty" may not be ideal here.